cpp/cmake/Dependencies.cmake

```diff
@@ -162,41 +162,52 @@ else()
 	set(FAISS_INSTALL_DIR ENV{FAISS_ROOT})
 	find_package(FAISS REQUIRED)
 endif(BUILD_STATIC_FAISS)
+
 ##############################################################################
 # - treelite build -----------------------------------------------------------
 
 find_package(Treelite 0.92 REQUIRED)
 
 ##############################################################################
-# - googletest ---------------------------------------------------------------
-
-set(GTEST_DIR ${CMAKE_CURRENT_BINARY_DIR}/googletest CACHE STRING
-  "Path to googletest repo")
-set(GTEST_BINARY_DIR ${PROJECT_BINARY_DIR}/googletest)
-set(GTEST_INSTALL_DIR ${GTEST_BINARY_DIR}/install)
-set(GTEST_LIB ${GTEST_INSTALL_DIR}/lib/libgtest_main.a)
-include(ExternalProject)
-ExternalProject_Add(googletest
-  GIT_REPOSITORY    https://github.com/google/googletest.git
-  GIT_TAG           6ce9b98f541b8bcd84c5c5b3483f29a933c4aefb
-  PREFIX            ${GTEST_DIR}
-  CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-                    -DBUILD_SHARED_LIBS=OFF
-                    -DCMAKE_INSTALL_LIBDIR=lib
-  BUILD_BYPRODUCTS  ${GTEST_DIR}/lib/libgtest.a
-                    ${GTEST_DIR}/lib/libgtest_main.a
-  UPDATE_COMMAND    "")
-
-add_library(gtestlib STATIC IMPORTED)
-add_library(gtest_mainlib STATIC IMPORTED)
-
-set_property(TARGET gtestlib PROPERTY
-  IMPORTED_LOCATION ${GTEST_DIR}/lib/libgtest.a)
-set_property(TARGET gtest_mainlib PROPERTY
-  IMPORTED_LOCATION ${GTEST_DIR}/lib/libgtest_main.a)
-
-add_dependencies(gtestlib googletest)
-add_dependencies(gtest_mainlib googletest)
+# - googletest build -----------------------------------------------------------
+
+if(BUILD_GTEST)
+	set(GTEST_DIR ${CMAKE_CURRENT_BINARY_DIR}/googletest CACHE STRING
+	  "Path to googletest repo")
+	set(GTEST_BINARY_DIR ${PROJECT_BINARY_DIR}/googletest)
+	set(GTEST_INSTALL_DIR ${GTEST_BINARY_DIR}/install)
+	set(GTEST_LIB ${GTEST_INSTALL_DIR}/lib/libgtest_main.a)
+	include(ExternalProject)
+	ExternalProject_Add(googletest
+	  GIT_REPOSITORY    https://github.com/google/googletest.git
+	  GIT_TAG           6ce9b98f541b8bcd84c5c5b3483f29a933c4aefb
+	  PREFIX            ${GTEST_DIR}
+	  CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+	                    -DBUILD_SHARED_LIBS=OFF
+	                    -DCMAKE_INSTALL_LIBDIR=lib
+	  BUILD_BYPRODUCTS  ${GTEST_DIR}/lib/libgtest.a
+	                    ${GTEST_DIR}/lib/libgtest_main.a
+	  UPDATE_COMMAND    "")
+	
+	add_library(GTEST::GTEST STATIC IMPORTED)
+	add_library(GTESTMAIN::GTESTMAIN STATIC IMPORTED)
+	
+	set_property(TARGET GTEST::GTEST PROPERTY
+	  IMPORTED_LOCATION ${GTEST_DIR}/lib/libgtest.a)
+	set_property(TARGET GTESTMAIN::GTESTMAIN PROPERTY
+	  IMPORTED_LOCATION ${GTEST_DIR}/lib/libgtest_main.a)
+	
+	set(GTEST_INCLUDE_DIRS "${GTEST_DIR}")
+	
+	add_dependencies(GTEST::GTEST googletest)
+	add_dependencies(GTESTMAIN::GTESTMAIN googletest)
+	
+else()
+	set(GTEST_INSTALL_DIR ENV{GTEST_ROOT})
+	set(GTESTMAIN_INSTALL_DIR ENV{GTEST_ROOT})
```

Comment:
Should probably do this after the `find_package` calls as those are what set `GTEST_ROOT` IIRC

Reply:
AFAICT, these variables are not actually needed. I was just setting them for convenience to match the manual build variables in the case where the user sets `GTEST_ROOT` manually.